spec.html

```diff
@@ -23352,6 +23351,12 @@ <h1>NaN</h1>
       <h1>undefined</h1>
       <p>The value of `undefined` is *undefined* (see <emu-xref href="#sec-ecmascript-language-types-undefined-type"></emu-xref>). This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *false* }.</p>
     </emu-clause>
+
+    <emu-clause id="sec-global-this">
+      <h1>globalThis</h1>
+      <p>The initial value of `globalThis` is the field of the current Realm record named [[GetGlobalThis]].
```

Comment:
So, there is no current Realm record during initialization. https://tc39.github.io/ecma262/#sec-global-object explicitly states there is no running execution context, and thus no current Realm record.

Additionally, there is no [[GetGlobalThis]] field on the current Realm record.

Reply:
hm, ok - i can clean up my typo, but given that there's no running execution context, how would you suggest i get to the global this value?

Reply:
My suggestion was "the Realm", alluding to the fact that each global has an associated Realm, and thus assuming we can reference it while defining properties of that global. That's the best I've got.

Reply:
i think it would make the most sense to explicitly set this in `SetRealmGlobalObject`

Reply:
Hmm yeah, doing things imperatively would be nice, I just don't see it being done for any other global properties. I guess you could do so and have this section be a reference to that.

Reply:
@devsnek it’s already set there, unless you mean it’d define it on the global object there? It seems strange to be mutating the global object or this value provided to that abstract op :-/

Reply:
i mean something along the lines of "the initial value of globalThis is undefined" (to match the initial slot values), then in `SetRealmGlobalObject` you'd do a Set() operation

Reply:
right, but someone might be passing a custom global object, or global this, to `SetRealmGlobalObject`, and it'd be weird to force it to have that property, no?

Reply:
hmm - functions have a `[[Realm]]` slot, so `%Function%.[[Realm]]` seems like it'd be available - however, i can't seem to figure out when builtin functions are created (https://tc39.github.io/ecma262/#sec-functioncreate which calls https://tc39.github.io/ecma262/#sec-functionallocate). Any tips?

Reply:
Most built-in functions are created in [CreateIntrinsics](https://tc39.github.io/ecma262/#sec-createintrinsics) via calls (explicit or implicit) to `CreateBuiltinFunction`.

Reply:
@domenic:
>  https://tc39.github.io/ecma262/#sec-global-object explicitly states there is no running execution context, and thus no current Realm record.

Not quite. It says that the global object "is created before control enters any execution context" (call this statement X). The spec doesn't really define what it means for control to enter an execution context; you've interpreted it as "when the context becomes the running execution context", and that's reasonable, but it has to occur later if X is to be true. (Practically speaking, X dates back to the first edition, long before there were realms, so you could imagine that the global object existed ab initio.)

In [InitializeHostDefinedRealm](https://tc39.github.io/ecma262/#sec-initializehostdefinedrealm), step 6 pushes a new context onto the execution context stack, making it the running execution context. At that point, `the current Realm Record` also becomes defined. (The new context's Realm component was set to the new realm at step 4.)

The object that will become the realm's global object is created in step 7 or 9, and then it is made the realm's `[[GlobalObject]]` in step 9.

So, when the global object is created, there *is* a `running execution context`, and there *is* a `current Realm Record`.

Reply:
oh awesome, in that case this approach should work?